locale.c

```diff
@@ -3830,10 +3830,19 @@ S_new_ctype(pTHX_ const char *newctype, bool force)
         DEBUG_L(PerlIO_printf(Perl_debug_log,
                               "Unsupported, MB_CUR_MAX=%d\n", mb_cur_max));
 
-        Perl_ck_warner_d(aTHX_ packWARN(WARN_LOCALE),
-                         "Locale '%s' is unsupported, and may crash the"
-                         " interpreter",
-                         newctype);
+        if (! IN_LC(LC_CTYPE) || ckWARN_d(WARN_LOCALE)) {
+            char * msg = Perl_form(aTHX_
+                                   "Locale '%s' is unsupported, and may hang"
+                                   " or crash the interpreter",
+                                     newctype);
+            if (IN_LC(LC_CTYPE)) {
+                Perl_warner(aTHX_ packWARN(WARN_LOCALE), "%s", msg);
+            }
+            else {
+                PL_warn_locale = newSV(0);
+                sv_setpvn(PL_warn_locale, msg, strlen(msg));
+            }
+    }
```

Comment:
This indent looks suspect - maybe just needs whitespace adjusting

Reply:
fixed indentation and updated